src/components/result-view-tab.tsx

```diff
@@ -0,0 +1,310 @@
+                isUnavailable: isUnavailableService(
+                    unavailableOptionalServices,
+                    OptionalServicesNames.SensitivityAnalysis
+                ),
+                enableDeveloperMode: true,
+                renderResult: renderSensitivityAnalysisResult,
+            },
+            {
+                id: 'ShortCircuit',
+                label: 'ShortCircuitAnalysis',
+                isUnavailable: isUnavailableService(
+                    unavailableOptionalServices,
+                    OptionalServicesNames.ShortCircuit
+                ),
+                enableDeveloperMode: enableDeveloperMode,
+                renderResult: renderShortCircuitAnalysisResult,
+            },
+            {
+                id: 'DynamicSimulation',
+                label: 'DynamicSimulation',
+                isUnavailable: isUnavailableService(
+                    unavailableOptionalServices,
+                    OptionalServicesNames.DynamicSimulation
+                ),
+                enableDeveloperMode: enableDeveloperMode,
+                renderResult: renderDynamicSimulationResult,
+            },
+            {
+                id: 'VoltageInit',
+                label: 'VoltageInit',
+                isUnavailable: isUnavailableService(
+                    unavailableOptionalServices,
+                    OptionalServicesNames.VoltageInit
+                ),
+                enableDeveloperMode: enableDeveloperMode,
+                renderResult: renderVoltageInitResult,
+            },
+        ];
+    }, [
+        unavailableOptionalServices,
+        enableDeveloperMode,
+        renderDynamicSimulationResult,
+        renderSecurityAnalysisResult,
+        renderSensitivityAnalysisResult,
+        renderShortCircuitAnalysisResult,
+        renderVoltageInitResult,
+        renderLoadFlowResult,
+    ]);
+
+    const renderTab = (service: IService) => {
+        return (
+            !service.isUnavailable &&
+            service.enableDeveloperMode && (
+                <Tab
+                    key={service.id + 'tab'}
+                    label={intl.formatMessage({
+                        id: service.label,
+                    })}
+                    disabled={disabled}
+                />
+            )
+        );
+    };
+    const renderTabPanelLazy: (service: IService) => React.ReactElement = (
+        service: IService
+    ) => {
+        return (
+            <>
+                {!service.isUnavailable && (
+                    <TabPanelLazy
+                        key={service.id + 'tabPanel'}
+                        className={classes.tabPanel}
+                        selected={!disabled}
+                    >
+                        {service.renderResult}
+                    </TabPanelLazy>
+                )}
+            </>
+        );
+    };
+
+    const selectedService = useMemo(() => {
+        const displayedServices: IService[] = services.filter(
```

Comment:
```suggestion
        const displayedServices = services.filter(
```

Try to use type inference when possible